src/client/java/teammates/client/scripts/scalabilitytests/InstructorFeedbackResultsPageScalabilityTest.java

```diff
@@ -18,28 +24,145 @@
 
     private static final String DATA_FOLDER_PATH = "src/client/java/teammates/client/scripts/scalabilitytests/data/";
     private static final Logger log = Logger.getLogger();
+    private static DataBundle testDataMax;
 
     @Override
     protected void prepareTestData() {
         // the actual test data is refreshed before each test method
     }
 
     private void refreshTestData(int numStudents, int numQuestions) {
-        refreshTestData(
-                "InstructorFeedbackResultsPageScaleTest-" + numStudents + "Students" + numQuestions + "Questions.json");
-    }
 
-    private void refreshTestData(String filename) {
-        testData = loadDataBundle(DATA_FOLDER_PATH + filename);
+        // modify set of students for each test according to its requirements
+        if (testData.students.size() == 0) { // if current collection of students is empty
+            addStudents(numStudents);
+        } else if (testData.students.size() < numStudents) { // current number of students is below required
+            addStudents(numStudents - testData.students.size());
+        } else if (testData.students.size() > numStudents) { // current number of students is above required
+            decreaseNumOfStudents(numStudents);
+        }
+
+        // modify set of questions for each test according to its requirements
+        if (testData.feedbackQuestions.size() == 0) { // if current collection of questions is empty
+            addQuestions(numQuestions);
+        } else if (testData.feedbackQuestions.size() < numQuestions) { // current number of questions is below required
+            addQuestions(numQuestions - testData.feedbackQuestions.size());
+        } else if (testData.feedbackQuestions.size() > numQuestions) { // current number of questions is above required
+            decreaseNumOfQuestions(numQuestions);
+        }
+
+        // modify set of responses for each test according to its sets of students and questions
+        updateFeedbackResponses();
+
         removeAndRestoreDataBundle(testData);
```

Comment:
Friends, please help with one more question: there is method ```verifyPresentInDatastore(DataBundle data)``` and inside it calls ```verifyAbsentInDatastore(EntityAttributes entity)``` only for some of the inclusive objects from DataBundle: ```accounts```, ```instructors```, ```courses``` and ```students```. While in our scalability test we read and write DataBundle with more inclusive object, like ```feedbackQuestions``` and ```feedbackResponses```. So ```verifyPresentInDatastore(DataBundle data)``` check separately for ```account```, ```instructor```, ```course``` and ```student``` if it exists in Datastore, but do not checks their connection as parts of some single DataBundle. 
And in our test method ```removeAndRestoreDataBundle(testData)``` reads and writes one single complete set of data. In this case: 
- do we necessarily want to test reading whole DataBundle from Datastore? Then seems like I need to create another overloaded method ```verifyPresentInDatastore(DataBundle data)``` to check all the objects in my DataBundle if they persist on Datastore. And if somehow only one entity is absent, progra,, will be forced to read and write whole DataBundle (same current number of operatios will remain)
- or can I modify the test and make it read only ```Students```, ```FeedbackQuestions``` and ```FeedbackResponses``` (data, which is actually scaled). In this case I will use current methods from ```Backdoor``` class like ```getStudent()``` and ```getFeedbackQuestion()``` to read ```Students```, ```FeedbackQuestions``` and ```FeedbackResponses```, which persisit in Datastore. In this case rest of objects from testData will not be tested.
Which is preferrable?